packages/account-tree-controller/src/AccountTreeController.ts

```diff
@@ -207,14 +250,39 @@ export class AccountTreeController extends BaseController<
   #handleAccountAdded(account: InternalAccount) {
     this.update((state) => {
       this.#insert(state.accountTree.wallets, account);
+
+      const context = this.#accountIdToContext.get(account.id);
+      if (context) {
+        const { walletId, groupId } = context;
+
+        const wallet = state.accountTree.wallets[walletId];
+        if (wallet) {
+          const walletInstance = this.getWallet(wallet.id);
+          if (walletInstance) {
+            if (wallet.metadata.name === '') {
+              this.#renameAccountWallet(walletInstance, wallet);
+            }
+
+            const group = wallet.groups[groupId];
+            if (group) {
+              const groupInstance = walletInstance.getAccountGroup(group.id);
+              if (groupInstance) {
+                if (group.metadata.name === '') {
+                  this.#renameAccountGroup(groupInstance, group);
+                }
+              }
+            }
+          }
```

Comment:
This feels really wrong to me to have all those `if`. Another alternative would be to `throw` if we cannot get the wallet/group instance. Normally instances and the state are always in sync, so this should be safe.